gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiKeyServiceImpl.java

```diff
@@ -137,6 +137,9 @@ public ApiKeyEntity renew(ExecutionContext executionContext, SubscriptionEntity
 
     @Override
     public ApiKeyEntity renew(ExecutionContext executionContext, SubscriptionEntity subscription, String customApiKey) {
+        if (subscription.getStatus().equals(io.gravitee.rest.api.model.SubscriptionStatus.CLOSED)) {
```

Comment:
I think you can only renew an ACCEPTED apikey.
```suggestion
        if (!subscription.getStatus().equals(io.gravitee.rest.api.model.SubscriptionStatus.ACCEPTED)) {
```